addons/binding/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/SpotifySession.java

```diff
@@ -0,0 +1,1299 @@
+        Gson gson = new Gson();
+        SpotifyWebAPIPlayerInfo playerInfo = gson.fromJson(result, SpotifyWebAPIPlayerInfo.class);
+        return playerInfo;
+    }
+
+    /*
+     * Callable Tasks for the Web API
+     * This has been created to setup the executor thread pool and ensure sequential processing of the Web API request
+     * tasks.
+     */
+    private class WebAPICallTask implements Callable<String> {
+
+        private String method;
+        private String url;
+        private String requestData;
+
+        public WebAPICallTask(String method, String url, String requestData) {
+            this.method = method;
+            this.url = url;
+            this.requestData = requestData;
+        }
+
+        @Override
+        public String call() throws Exception {
+            return callWebAPI();
+        }
+
+        /**
+         * This method is a simple wrapper for Spotify Web API calls
+         *
+         * @param method the http method to use (GET, PUT, POST ..)
+         * @param url the Web API url to call
+         * @param requestData the body of the request, if any.
+         * @return response from call
+         */
+        private String callWebAPI() {
+            logger.trace("Calling Spotify Web API at {}:{} with:\n{}", method, url, requestData);
+            Properties headers = new Properties();
+            headers.setProperty("Authorization", "Bearer " + accessToken);
+            headers.setProperty("Accept", "application/json");
+
+            String contentType = "application/json";
+            ContentResponse response = null;
+
+            // TODO: manage http timeout exceptions in a better way, currently crash the polling thread
+            for (int i = 0; i < HTTP_CLIENT_RETRY_COUNT; i++) {
+                logger.debug("Spotify Web API call attempt: {}", i + 1);
+                try {
+                    response = httpClient.newRequest(url).method(HttpMethod.fromString(method))
+                            .header("Authorization", "Bearer " + accessToken).header("Accept", "application/json")
+                            .content(new StringContentProvider(requestData), contentType)
+                            .timeout(HTTP_CLIENT_TIMEOUT, TimeUnit.SECONDS).send();
+
+                    logger.debug("Response Code: {}", response.getStatus());
+                    logger.trace("Response Data: {}", response.getContentAsString());
+
+                    // Response Code 429 means requests rate limits exceeded.
+                    if (response.getStatus() == 429) {
+                        String retryAfter = response.getHeaders().get("Retry-After");
+                        logger.warn(
+                                "Spotify Web API returned code 429 (rate limit exceeded). Retry After {} seconds. Decrease polling interval of bridge! Going to sleep...",
+                                retryAfter);
+
+                        Thread.sleep(Integer.parseInt(retryAfter));
```

Comment:
Thead sleeps is most of the time not the solution, the problem we are trying to resolve is that we should processing the queue for a certain time? Maybe a different setup is needed for that?